File: Backend-RS/models/menciones_usuario.py
# models/menciones_usuario.py
from sqlmodel import Field, SQLModel, Relationship
from typing import Optional
import datetime
from models.publicaciones import Publicaciones  
from models.usuario import Usuario
from config.database import get_session

class MencionesUsuarioBase(SQLModel):
    Fecha: datetime.datetime = Field(default_factory=datetime.datetime.utcnow)

class MencionesUsuario(MencionesUsuarioBase, table=True):
    IdMencion: Optional[int] = Field(default=None, primary_key=True)
    IdPublicacion: int = Field(foreign_key="publicaciones.IdPublicacion")
    IdUsuario: int = Field(foreign_key="usuario.IdUsuario")

    publicacion: "Publicaciones" = Relationship(back_populates="menciones_usuario")
    usuario: "Usuario" = Relationship(back_populates="menciones_recibidas")

class MencionesUsuarioCreate(MencionesUsuarioBase):
    IdPublicacion: int
    IdUsuario: int

class MencionesUsuarioPublic(MencionesUsuarioBase):
    IdMencion: int
    IdPublicacion: int
    IdUsuario: int

class MencionesUsuarioUpdate(MencionesUsuarioBase):
    pass

MencionesUsuarioPublic.model_rebuild()

File: frontendrs/lib/data/models/menciones_usuario.dart
class MencionesUsuario {
  final int idMencion;
  final int idPublicacion;
  final int idUsuario;
  final DateTime fecha;

  const MencionesUsuario({
    required this.idMencion,
    required this.idPublicacion,
    required this.idUsuario,
    required this.fecha,
  });

  factory MencionesUsuario.fromJson(Map<String, dynamic> json) {
    return MencionesUsuario(
      idMencion: json['IdMencion'] ?? 0,
      idPublicacion: json['IdPublicacion'] ?? 0,
      idUsuario: json['IdUsuario'] ?? 0,
      fecha: DateTime.parse(json['Fecha']),
    );
  }

  Map<String, dynamic> toJson() => {
    'IdMencion': idMencion,
    'IdPublicacion': idPublicacion,
    'IdUsuario': idUsuario,
    'Fecha': fecha.toIso8601String(),
  };
}

class MencionesUsuarioCreate {
  final int idPublicacion;
  final int idUsuario;

  const MencionesUsuarioCreate({
    required this.idPublicacion,
    required this.idUsuario,
  });

  Map<String, dynamic> toJson() => {
    'IdPublicacion': idPublicacion,
    'IdUsuario': idUsuario,
  };
}

class MencionesUsuarioUpdate {
  final DateTime fecha;

  const MencionesUsuarioUpdate({required this.fecha});

  Map<String, dynamic> toJson() => {'Fecha': fecha.toIso8601String()};
}
